packages/xo-server/src/xapi/index.mjs

```diff
@@ -566,8 +566,8 @@ export default class Xapi extends XapiBase {
     for (const vbd of vbds) {
       if (vbd.type === 'Disk') {
         const vdi = vbd.$VDI
-        // Ignore VDI snapshots which have a parent
-        if (vdi.$snapshot_of !== undefined) {
+        // We need to be sure it's not a snapshot (prevent VDI_NOT_IN_MAP error)
```

Comment:
The previous test already implicitly tested that it was a snapshot, the comment must explain why the test had to be made explicit.

Reply:
I don't know how else to explain it. The change was made just to fix the bug. What do you suggest?

Reply:
As I explained to you, there is no bug in XO's code.

The problem comes from XAPI itself (observed on an XCP-ng 8.1 host, might be related to CBT), where some non-snapshot VDIs had a `snapshot_of` property not equal to `OpaqueRef:NULL` as expected but containing a reference to an existing VDI.

Also, please include the link of this [internal discussion](https://team.vates.fr/vates/pl/9m4u5rsr7tfcdroykwq6i6mcmo), even it is private, I believe it's worth including it.

Reply:
Done